### ISSUE TITLE
Snap fails to run under Solus

### ISSUE THREAD

**Issue by thornbill** [wavebox/waveboxapp#521]
* Which version of Wavebox are you using? [e.g. 3.0.0]
  * 3.7.0
* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
  * Solus

When attempting to run the snap under Solus, the application fails to launch. This is the terminal output from running `wavebox`.

```
➔ wavebox
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"
Gtk-Message: Failed to load module "pk-gtk-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
[10365:1220/122314.003086:FATAL:platform_font_linux.cc(63)] Check failed: typeface. Could not find any font: Noto Sans, sans
#0 0x000000b1e9d7 <unknown>
#1 0x000000ad618b <unknown>
#2 0x000002cdb29f <unknown>
#3 0x000002cdaedf <unknown>
#4 0x000002cdbc06 <unknown>
#5 0x000002cd4129 <unknown>
#6 0x000001e31400 <unknown>
#7 0x000001ec62d3 <unknown>
#8 0x000003eed08d atom::api::WebContents::InitWithSessionAndOptions()
#9 0x000003eed89f atom::api::WebContents::WebContents()
#10 0x000003ef9236 atom::api::WebContents::Create()
#11 0x000003f0b4ba atom::api::Window::Window()
#12 0x000003f0d0bc atom::api::Window::New()
#13 0x000003eacde0 mate::internal::InvokeNew<>()
#14 0x000003eacf70 mate::internal::Dispatcher<>::DispatchToCallback()
#15 0x7f0ea7803429 <unknown>
#16 0x7f0ea77e6ccc <unknown>
#17 0x7f0ea77e6706 <unknown>
#18 0x1b2ad228437d <unknown>

[1]    10365 abort      wavebox
```

**Comment by thornbill** [comment]
There is a similar issue reported on the [snapcraft forum](https://forum.snapcraft.io/t/could-not-find-any-font-error-on-solus/3251).

**Comment by tonyjames** [comment]
I'm allso experiencing this issue with Fedora 27:

$ snap run wavebox
Gtk-Message: Failed to load module "pk-gtk-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
[8191:1220/192246.701650:FATAL:platform_font_linux.cc(63)] Check failed: typeface. Could not find any font: Cantarell, sans
#0 0x000000b1e9d7 <unknown>
#1 0x000000ad618b <unknown>
#2 0x000002cdb29f <unknown>
#3 0x000002cdaedf <unknown>
#4 0x000002cdbc06 <unknown>
#5 0x000002cd4129 <unknown>
#6 0x000001e31400 <unknown>
#7 0x000001ec62d3 <unknown>
#8 0x000003eed08d atom::api::WebContents::InitWithSessionAndOptions()
#9 0x000003eed89f atom::api::WebContents::WebContents()
#10 0x000003ef9236 atom::api::WebContents::Create()
#11 0x000003f0b4ba atom::api::Window::Window()
#12 0x000003f0d0bc atom::api::Window::New()
#13 0x000003eacde0 mate::internal::InvokeNew<>()
#14 0x000003eacf70 mate::internal::Dispatcher<>::DispatchToCallback()
#15 0x7fe52ea04429 <unknown>
#16 0x7fe52e9e7ccc <unknown>
#17 0x7fe52e9e7706 <unknown>
#18 0x3e7f8b38437d <unknown>

Aborted (core dumped)

**Comment by Thomas101** [comment]
@thornbill that was me on the forum. We've done a fair bit of head-scratching to try and get this working, but it looks like some of the dependencies that snap include or the Solus integration is broken. I've been able to reproduce the issue here, but not been able to find a fix quite yet. I've also tried updating (**not recommended!**) snap to 2.29 on our Solus box but this stops all snaps from running, so I wouldn't recommend trying that.

So you can get back up and running, there's a couple of things you can try...

1. Install the tar version from https://wavebox.io/download. This should work out of the box for you, although you'll need to drag it to /opt/wavebox or similar
2. I've uploaded the 3.6.0 snap version to our server. If you want to download that and install you can by doing the following....
    * Make a copy of your user data. Some info on where that's stored here... https://wavebox.io/kb/data-locations
    * Download the 3.6.0 snap from http://wavebox.io/dl/builds/Wavebox_r46_x86.snap
    * Run sudo snap remove wavebox
    * Run sudo snap install Wavebox_r46_x86.snap --devmode
    * Copy your user data back in

Once we've figured out what the upstream problem is, we'll update the live snap packages and push out an update

**Comment by Ads20000** [comment]
> I've also tried updating (not recommended!) snap to 2.29 on our Solus box but this stops all snaps from running, so I wouldn't recommend trying that.

What do you mean by this? `snapd` is already at 2.29.4.2, do you mean 2.30 (in which case that needs to be reported on the forum, that's quite a big problem which needs to be resolved given 2.30 is due to be released very soon (technically 18th December))?

**Comment by Thomas101** [comment]
2.29.2 is what's broken for me. Reported on the Solus forum but waiting for admin to approve my first post.

I'm guessing by that you're already running 2.29 on Solus? If so, it may just be my test box

**Comment by Ads20000** [comment]
Oh, no I'm on Ubuntu, I thought the `core` snap (which should be the same version as `snapd`) was the same version on all distros because it re-execs the new version via an automated `snap refresh`? Perhaps @ikeydoherty disabled that on Solus or something?

Sorry, I'm a 'Regular' on the forums so I keep a fairly close eye on snappy but perhaps I don't know as much as I should :(

**Comment by Thomas101** [comment]
@Ads20000 As regards updating to snap 2.29 on Solus, looks like you need a kernel update to go with it.

You may be right on the core snap. I think the integration points are different for different distros though. We've had snaps broken on certain versions of Ubuntu, different groups of distros etc.

As for this issue though, it's definitely limited to some distros. We've spent three days trying to get to the bottom of this so far and no juice. We managed to get the app running by including the `desktop` plug, but this means font's don't render at all. We've even tried re-packing Wavebox 3.6.0 but this comes up with the same issue, so it's definitely one of the dependencies snap pulls in.

**Comment by ikeydoherty** [comment]
Not sure why you believe you need to upgrade snapd or the kernel. We have `4.14.6` in shannon which is newer than Ubuntu, and if you manually update snapd without our patch set or configuration you will cause snapd to break.

snapd is now at 2.30 in unstable and will be released to shannon tomorrow.

Now, what is **more likely** is that you're using the new desktop helper system and expecting the new font integration to work, which is actually mostly in git master and not in release, but **is** in the `core` version of snapd in Ubuntu :)

I would say try again tomorrow with new snapd and all the other updates so we're all on the same page

**Comment by Thomas101** [comment]
Just thought we'd give updating a try - it's fixed issues with snap in the past.

We weren't originally using the desktop plug nor changed our snapcraft.yaml file for some time which is why we thought the breakage might be with snap or one of its dependencies. Since we've had this issue we have tried adding the desktop plug as a similar project mentioned that adding it helped but to no avail.

Is there a specific time that 2.30 is meant to be going out tomorrow?

**Comment by ikeydoherty** [comment]
We'll have it out on unstable for around 5PM GMT/IST - and once we all have some equal footing to test with I can quite happily help you debug this and if its a Solus problem, unfudge Solus :)

**Comment by Thomas101** [comment]
After a bit of advice from the snap team we've managed to get a version together that works on Fedora & Solus. We've pushed this out to the edge channel for testing. If you want to give it a try you can use the following from the terminal...

* To install `snap install wavebox --channel=edge`
* To update `snap refresh wavebox --channel=edge`

You may also need to install some additional fonts on your machine so that Wavebox looks the part. We've just added a [KB article](https://wavebox.io/kb/installing-linux-fonts) about this to get you started.

Would be great to know how this works for everyone!
Thanks

**Comment by pm314** [comment]
Edge channel version works great on latest Solus. Looks much better that 3.6 also! Thanks for addressing.

**Comment by thornbill** [comment]
It is working well for me on Solus also. Thanks!

**Comment by bradstewart** [comment]
Working on Fedora 27 as well.

**Comment by Thomas101** [comment]
Thanks for the feedback! We've just released this out to the beta channel with 3.7.3. All being well we'll be pushing out to stable early next week :)

**Comment by Thomas101** [comment]
Hi, this has just been released in the latest version of Wavebox (3.8.0) and to the stable snap channel. Thanks!
